os/mm/mm_heap/mm_getsize.c

```diff
@@ -90,3 +90,37 @@ size_t mm_get_largest_freenode_size(void)
 
 	return largest_size;
 }
+
+/****************************************************************************
+ * Name: mm_get_heap_free_size
+ *
+ * Description:
+ *   returns free size in the heap
+ ****************************************************************************/
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+size_t mm_get_heap_free_size(void)
+{
+	struct mm_heap_s *heap;
+
+#ifdef __KERNEL__
+	size_t free_size = 0;
+	int heap_idx;
+	heap = g_kmmheap;
+	struct mm_heap_s *target_heap;
+
+	/* Kernel can have multi-heap, so traversing all n-heaps and add all free sizes. */
+	for (heap_idx = HEAP_START_IDX; heap_idx <= HEAP_END_IDX; heap_idx++) {
+		target_heap = &heap[heap_idx];
+		free_size += (target_heap->mm_heapsize - target_heap->total_alloc_size);
+	}
```

Comment:
There is no return value in this #ifdef.

Reply:
Added. Thanks.

Reply:
Not added. Please check it again.

Reply:
Added. Thanks.